cmd/otelcol/config/collector/config.d.linux/receivers/mongodb.discovery.yaml

```diff
@@ -0,0 +1,62 @@
+# in discovery mode (--discovery) and is provided for reference or customization.   #
+# Please note that any changes made to this file will need to be reconciled during  #
+# upgrades of the Collector.                                                        #
+#####################################################################################
+# mongodb:
+#   enabled: true
+#   rule:
+#     docker_observer: type == "container" and any([name, image, command], {# matches "(?i)mongo"}) and not (command matches "splunk.discovery")
+#     host_observer: type == "hostport" and command matches "(?i)mongo" and not (command matches "splunk.discovery")
+#     k8s_observer: type == "port" and pod.name matches "(?i)mongo"
+#   config:
+#     default:
+#       username: splunk.discovery.default
+#       password: splunk.discovery.default
+#       tls:
+#         insecure: true
+#         insecure_skip_verify: true
+#   status:
+#     metrics:
+#       - status: successful
+#         strict: mongodb.database.count
+#         log_record:
+#           body: mongodb receiver is working!
+#     statements:
+#       - status: failed
+#         regexp: 'connect: network is unreachable'
+#         log_record:
+#           append_pattern: true
+#           body: The container cannot be reached by the Collector. Make sure they're in the same network.
+#       - status: failed
+#         regexp: 'connect: connection refused'
+#         log_record:
+#           append_pattern: true
+#           body: The container is refusing mongodb connections.
+#       - status: partial
+#         regexp: '.* unable to authenticate using mechanism .*'
+#         log_record:
+#           append_pattern: true
+#           body: >-
+#               Please ensure your user credentials are correctly specified with
+#               `--set splunk.discovery.receivers.mongodb.config.username="<username>"` and
+#               `--set splunk.discovery.receivers.mongodb.config.password="<password>"` or
+#               `SPLUNK_DISCOVERY_RECEIVERS_mongodb_CONFIG_username="<username>"` and
+#               `SPLUNK_DISCOVERY_RECEIVERS_mongodb_CONFIG_password="<password>"` environment variables.
+#       - status: partial
+#         regexp: '.* failed to fetch index stats metrics: (Unauthorized) not authorized on admin to execute command .*'
+#         log_record:
+#           append_pattern: true
+#           body: >-
+#             Make sure the user has been given access admin db with full root access.
+#             `db.grantRolesToUser('admin', [{ role: 'readWrite', db: 'admin' }])`
+#       - status: partial
+#         regexp: '.* error: invalid configuration: receivers::mongodb: invalid transport type .*'
+#         log_record:
+#           append_pattern: true
+#           body: >-
+#             Make sure to have a valid transport type inside the otel configuration. For more information check the documentation carefully.
+#             `transport: tcp`
```

Comment:
This one is unclear on how a user should resolve it. We don't provide a config interface for particular components with the discovery mode

Reply:
I think we can remove it. With the default configuration, we cannot hit this scenario. Only possible if someone is manual configuring but not in discovery mode.